### PR TITLE
sensors: ist8310: Add Mag driver

### PR DESCRIPTION
**Pull request by PetervdPerk-NXP**
Adds support for the Isentek IST8310 3-axis magnetic sensor

### CODE REVIEW

File: drivers/sensor/ist8310/ist8310.h
Comment (yperess) [inline drivers/sensor/ist8310/ist8310.h:51]: nit: duplicate includes from above and sort order

**Review by yperess — CHANGES_REQUESTED** [review]
Need to add tests for this under https://github.com/zephyrproject-rtos/zephyr/tree/main/tests/drivers/build_all/sensor.

@tristan-google can help as he's working on a catch all test for sensors.

**Comment by PetervdPerk-NXP** [comment]
Does just adding the following suffice, or is more needed?
```
test_i2c_ist8310@6f {
	compatible = "isentek,ist8310";
	reg = <0x6f>;
	status = "okay";
};
```

**Comment by tristan-google** [comment]
> Does just adding the following suffice, or is more needed?

Yes, that should cause the driver to build since you set up Kconfig to enable that sensor based on the device tree. I would run `scripts/twister -civ -T tests/drivers/build_all/sensor/` to ensure no compilation errors and check the output directory or logs to verify `ist8310.c` actually built. Note that without a matching emulator, this test will really only be checking that the driver builds and links (Zephyr will attempt to initialize it, but that will gracefully fail at the first I2C transaction since there is nothing on the other end --this is OK)

**Comment by PetervdPerk-NXP** [comment]
Thanks building locally using `scripts/twister -civ -T tests/drivers/build_all/sensor/` is working and CI output looks okay as well.

**Comment by bperseghetti** [comment]
@tristan-google @yperess @teburd all good now to approve/merge?

**Review by bperseghetti — APPROVED** [review]

**Comment by igalloway** [comment]
@yperess  - are you ok to approve now that the test has been added?

File: drivers/sensor/ist8310/ist8310.c
Comment (yperess) [inline drivers/sensor/ist8310/ist8310.c:66]: I think these are missing a call to `sys_be16_to_cpu`. Should look like:

```c
drv_data->sample_x = sys_be16_to_cpu((buff[1] << 8) | buff[0]);
```

File: tests/drivers/build_all/sensor/i2c.dtsi
Comment (yperess) [inline tests/drivers/build_all/sensor/i2c.dtsi:851]: (I won't block on this one)

This is a good start to make sure that the driver builds at least, ideally we'd see an emulator with the driver (not yet a hard requirement in Zephyr). You can see https://github.com/zephyrproject-rtos/zephyr/tree/main/drivers/sensor/akm09918c as a sample of a similar device's emulator.

**Review by yperess — CHANGES_REQUESTED** [review]
I think there's a small bug in the way the driver was storing the values into `sample_*`, other than that just run `clang-format` and it looks ready. (would love to see an emulator with this to make sure the driver doesn't regress).

File: drivers/sensor/ist8310/ist8310.c
Comment (PetervdPerk-NXP) [reply drivers/sensor/ist8310/ist8310.c]: Good point missed that one, made it more portable now.

File: tests/drivers/build_all/sensor/i2c.dtsi
Comment (bperseghetti) [reply tests/drivers/build_all/sensor/i2c.dtsi:851]: Awesome thanks, we will look to add the emulation down the road with some other sensors as well.

**Review by teburd — APPROVED** [review]

**Review by yperess — APPROVED** [review]

File: tests/drivers/build_all/sensor/i2c.dtsi
Comment (fabiobaltieri) [inline tests/drivers/build_all/sensor/i2c.dtsi:850]: this is not needed, nodes are enabled by default

File: drivers/sensor/ist8310/ist8310.h
Comment (fabiobaltieri) [inline drivers/sensor/ist8310/ist8310.h:97]: leftover? Can't find any reference to this

File: drivers/sensor/ist8310/ist8310.c
Comment (fabiobaltieri) [inline drivers/sensor/ist8310/ist8310.c:184]: no need to initialize this one

File: drivers/sensor/CMakeLists.txt
Comment (fabiobaltieri) [inline drivers/sensor/CMakeLists.txt:57]: if you meant to insert this alphabetically, you almost made it :-)

Comment (fabiobaltieri) [reply drivers/sensor/CMakeLists.txt:57]: if I meant to make this review before merge, I almost made it :-D

File: drivers/sensor/ist8310/ist8310.c
Comment (fabiobaltieri) [inline drivers/sensor/ist8310/ist8310.c:64]: you could drop the outer parenthesis here

**Comment by bperseghetti** [comment]
@fabiobaltieri I can open a different PR with those changes, or when we add the emulation for sensors with @tristan-google and @yperess work we could make those changes then.

**Comment by fabiobaltieri** [comment]
> @fabiobaltieri I can open a different PR with those changes, or when we add the emulation for sensors with @tristan-google and @yperess work we could make those changes then.

Either works, thanks.

**Pull request merged**
